ceno_zkvm/src/instructions/riscv/rv32im.rs

```diff
@@ -144,92 +144,163 @@ pub struct Rv32imConfig<E: ExtensionField> {
     pub ltu_config: <LtuTableCircuit<E> as TableCircuit<E>>::TableConfig,
     #[cfg(not(feature = "u16limb_circuit"))]
     pub pow_config: <PowTableCircuit<E> as TableCircuit<E>>::TableConfig,
+    // record InsnKind -> cells
+    pub inst_cells_map: Vec<u64>,
+    // record opcode name -> cells
+    // serve ecall/table for no InsnKind
+    pub ecall_table_cells_map: HashMap<String, u64>,
 }
 
 impl<E: ExtensionField> Rv32imConfig<E> {
     pub fn construct_circuits(cs: &mut ZKVMConstraintSystem<E>) -> Self {
+        let mut inst_cells_map = vec![0; InsnKind::COUNT];
```

Comment:
cell of each opcode are recording during circuit registration.
This avoid per opcode/table trait interface